lib/index.js

```diff
@@ -238,16 +247,50 @@ Builder.prototype.build = function(options) {
 			result.allVariables = oVariableCollector.getAllVariables();
 
 			// also compile rtl-version if requested
+			let oRTL;
 			if (options.rtl) {
-				result.cssRtl = tree.toCSS(assign(options.compiler, {
-					plugins: [new RTLPlugin()]
+				const RTLPlugin = require("./plugin/rtl");
+				oRTL = new RTLPlugin();
+			}
+
+			if (oRTL) {
+				result.cssRtl = tree.toCSS(assign({}, options.compiler, {
+					plugins: [oRTL]
 				}));
 			}
 
 			if (rootpath) {
 				result.imports.unshift(rootpath);
 			}
 
+			// also compile css-variables version if requested
+			if (options.compiler.cssVars) {
+				const CSSVariablesCollectorPlugin = require("./plugin/css-variables-collector");
+				const oCSSVariablesCollector = new CSSVariablesCollectorPlugin(config);
+				result.cssVars = {};
+				result.cssVars.css = tree.toCSS(assign({}, options.compiler, {
+					plugins: [oCSSVariablesCollector]
+				}));
+				result.cssVars.source = oCSSVariablesCollector.toLessVariables();
+				if (oRTL) {
+					const oCSSVariablesCollectorRTL = new CSSVariablesCollectorPlugin(config);
+					result.cssVars.cssRtl = tree.toCSS(assign({}, options.compiler, {
+						plugins: [oCSSVariablesCollectorRTL, oRTL]
+					}));
+				}
+				return new Promise(function(resolve, reject) {
+					// generate the CSS variables content out of the less variables
+					parser.parse(result.cssVars.source, function(err, tree) {
+						if (err) {
+							reject(err);
+						} else {
+							result.cssVars.vars = tree.toCSS(assign({}, options.compiler));
```

Comment:
would it make sense to rename `vars` to `cssVariables` since it will be the content of the variables.css?

Reply:
I adopt `vars` to `cssVariables` and `source` to `lessVariables`